### PR TITLE
[PT2.0+CUDA118] [Training GPU] Reduce Image size

### PR DESCRIPTION
**Pull request by jeet4320**
*GitHub Issue #, if available:*

**Note**: 
- If merging this PR should also close the associated Issue, please also add that Issue # to the Linked Issues section on the right. 

- All PR's are checked weekly for staleness. This PR will be closed if not updated in 30 days.

### Description

Reduce image size:
    - Install needed packages instead of cudatoolkit to reduce size of image
    - Use mamba clean after every mamba install to clear its cache

SM image reduction: 23.4-18=5.4 GB
EC2 image reduction: 20.8-15.5=5.3GB

### Tests run

**NOTE: By default, docker builds are disabled. In order to build your container, please update dlc_developer_config.toml and specify the framework to build in "build_frameworks"**
- [x] I have run builds/tests on commit <INSERT COMMIT ID> for my changes.

EC2, EKS, ECS, SM local, SM Remote: https://github.com/aws/deep-learning-containers/pull/3337/commits/f001ec77a2a5fbb3281ccc012db046efc83b4e55

Sanity tests pass: https://github.com/aws/deep-learning-containers/pull/3337/commits/e0a2e92638e36282b735fa505f642dd99f834773

SM EFA tests passed: https://github.com/aws/deep-learning-containers/pull/3337/commits/4616ac63fcac6449aa3c87af6cb37a5746a83bea

**NOTE: If you are creating a PR for a new framework version, please ensure success of the standard, rc, and efa sagemaker remote tests by updating the dlc_developer_config.toml file:**

- [ ] Revision A: `sagemaker_remote_tests = "standard"`
- [ ] Revision B: `sagemaker_remote_tests = "rc"`
- [ ] Revision C: `sagemaker_remote_tests = "efa"`

**Additionally, please run the sagemaker local tests in at least one revision:**
- [ ] `sagemaker_local_tests = true`

### Formatting
- [ ] I have run `black -l 100` on my code (formatting tool: https://black.readthedocs.io/en/stable/getting_started.html)

## PR Checklist
- [ ] I've prepended PR tag with frameworks/job this applies to : [mxnet, tensorflow, pytorch] | [ei/neuron/graviton] | [build] | [test] | [benchmark] | [ec2, ecs, eks, sagemaker]
- [ ] If the PR changes affects SM test, I've modified dlc_developer_config.toml in my PR branch by setting sagemaker_tests = true and efa_tests = true
- [ ] If this PR changes existing code, the change fully backward compatible with pre-existing code. (Non backward-compatible changes need special approval.)
- [ ] (If applicable) I've documented below the DLC image/dockerfile this relates to
- [ ] (If applicable) I've documented below the tests I've run on the DLC image
- [ ] (If applicable) I've reviewed the licenses of updated and new binaries and their dependencies to make sure all licenses are on the Apache Software Foundation Third Party License Policy Category A or Category B license list.  See [https://www.apache.org/legal/resolved.html](https://www.apache.org/legal/resolved.html).
- [ ] (If applicable) I've scanned the updated and new binaries to make sure they do not have vulnerabilities associated with them.

## Pytest Marker Checklist
- [ ] (If applicable) I have added the marker `@pytest.mark.model("<model-type>")` to the new tests which I have added, to specify the Deep Learning model that is used in the test (use `"N/A"` if the test doesn't use a model)
- [ ] (If applicable) I have added the marker `@pytest.mark.integration("<feature-being-tested>")` to the new tests which I have added, to specify the feature that will be tested
- [ ] (If applicable) I have added the marker `@pytest.mark.multinode(<integer-num-nodes>)` to the new tests which I have added, to specify the number of nodes used on a multi-node test
- [ ] (If applicable) I have added the marker `@pytest.mark.processor(<"cpu"/"gpu"/"eia"/"neuron">)` to the new tests which I have added, if a test is specifically applicable to only one processor type

#### EIA/NEURON/GRAVITON Testing Checklist
* When creating a PR:
- [ ] I've modified `dlc_developer_config.toml` in my PR branch by setting `ei_mode = true`, `neuron_mode = true` or `graviton_mode = true`

#### Benchmark Testing Checklist
* When creating a PR:
- [ ] I've modified `dlc_developer_config.toml` in my PR branch by setting `benchmark_mode = true`

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license. I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: dlc_developer_config.toml [hunk 1]
```diff
@@ -56,18 +56,18 @@ ec2_tests = true
...
-sagemaker_local_tests = false
+sagemaker_local_tests = true
 
 # SM remote test valid values:
 # "off" --> do not trigger sagemaker remote tests (default)
 # "standard" --> run standard sagemaker remote tests from test/sagemaker_tests
 # "rc" --> run release_candidate_integration tests
 # "efa" --> run efa sagemaker tests
-sagemaker_remote_tests = "off"
+sagemaker_remote_tests = "efa"
```
Comment (saimidu) [inline dlc_developer_config.toml:70]: Please also run the tests with this set to "standard".

File: test/dlc_tests/sanity/test_utility_installation.py [hunk 2]
```diff
@@ -101,6 +101,7 @@ def test_utility_packages_using_import(training):
             executable="python",
         ).stdout.strip()
         if package == "sagemaker":
+            version = version.splitlines()[-1]
```
Comment (saimidu) [inline test/dlc_tests/sanity/test_utility_installation.py:104]: If you are contributing this as a part of the PR, could you please change this to something like
https://github.com/aws/deep-learning-containers/pull/3293/files#diff-b38ca2feaf53682fc202ce43ba88a2d521bbfe77bddbd48040ce3562d42227ddR98-R107
with a regular expression to search for the version?

**Comment by saimidu** [comment]
Thanks for adding the commit-ids where the tests succeeded on the PR Description.

Could you also run the SM Remote EFA tests (set `sagemaker_remote_tests = "efa"`) for the CUDA 11.8 image?

File: pytorch/training/docker/2.0/py3/cu118/Dockerfile.gpu [hunk 3]
```diff
@@ -382,13 +392,78 @@ ARG PT_TORCHVISION_URL
 ARG PT_TORCHAUDIO_URL
 ARG PT_TORCHDATA_URL
 ARG PYTORCH_VERSION_SM
+ARG NCCL_ASYNC_ERROR_HANDLING
 
-# Install sm PyTorch (force remove will increase docker image size drastically)
+# Install ec2 PyTorch
```
Comment (jeet4320) [inline pytorch/training/docker/2.0/py3/cu118/Dockerfile.gpu:397]: comment to be changed

File: pytorch/training/docker/2.0/py3/cu118/Dockerfile.gpu [hunk 4]
```diff
@@ -401,15 +394,77 @@ ARG PT_TORCHDATA_URL
 ARG PYTORCH_VERSION_SM
 ARG NCCL_ASYNC_ERROR_HANDLING
 
-# Install sm PyTorch (force remove will increase docker image size drastically)
+# Install ec2 PyTorch
```
Comment (saimidu) [inline pytorch/training/docker/2.0/py3/cu118/Dockerfile.gpu:397]: Please change this to "sm" PyTorch after the tests all run and the PR has temporary changes removed.

**Review by saimidu — COMMENTED** [review]
Everything in the dockerfile/test changes looks good to approve. Waiting for tests to complete to approve the PR

**Comment by jeet4320** [comment]
newer version of accelerate released last week breaks pip check, so pinning it as this PR is created to reduce the size of image

**Pull request merged**